nvflare/apis/impl/task_controller.py

```diff
@@ -83,6 +83,7 @@ def broadcast_and_wait(
         request = task.data
         # apply task filters
         self.log_debug(fl_ctx, "firing event EventType.BEFORE_TASK_DATA_FILTER")
+        fl_ctx.set_prop(FLContextKey.TASK_DATA, task.data, sticky=False, private=True)
         self.fire_event(EventType.BEFORE_TASK_DATA_FILTER, fl_ctx)
```

Comment:
Is this dangerous?
If we set the task.data in the fl_ctx, then any components can use fire_event to modify it.
Will that cause a racing issue?
Will that corrupt our task?

Reply:
This is a double-edged sword (like in many other places). Even handlers could have legitimate reason to process or just read the task data. I keep this here to be consistent with the existing behavior of server/client task interactions.